config/src/lib.rs

```diff
@@ -997,47 +1006,11 @@ impl Config {
             paths.insert(0, path.into());
         }
 
-        for p in &paths {
```

Comment:
(see my comment down in main.rs before reading this one)

then, this logic could be kept over here with a minor addition being inserted here (github won't let me comment on lines that are not part of a PR; this logically belongs right after the insertion logic for the env var above so that this insert takes precedence at the start of the vec):

```rust
   if let Some(path) = CONFIG_FILE_OVERRIDE.lock().unwrap().as_ref() {
      paths.insert(0, path.clone());
   }
```

Reply:
I did this, but I think it's a bit wrong:
If the file given with `--config-file foo/bar` does NOT exist, the next possibility will be considered.
:point_right:  I would expect it to fail to load (and fallback to default config?) if the given file is unreadable, no?

Reply:
Good catch.  I think this a good place to introduce a little enum:

```rust
enum Item {
   Required(PathBuf),
   Optional(PathBuf),
}
```

Then the members of the `paths` vec here are `Item::Required` for entries produced from the `CONFIG_FILE_OVERRIDE` or the `WEZTERM_CONFIG_FILE` environment, and `Item::Optional` for all of the others.

Then in the loop below the `NotFound => continue` line would bail for `Required` and continue for `Optional`.

Reply:
see reply below https://github.com/wez/wezterm/pull/459#issuecomment-774679104